Source/WebKit/UIProcess/API/Cocoa/WKWebpagePreferences.mm

```diff
@@ -586,14 +586,31 @@ - (void)setLockdownModeEnabled:(BOOL)lockdownModeEnabled
 #endif
 }
 
-- (BOOL)_networkConnectionIntegrityEnabled
+- (_WKWebsiteNetworkConnectionIntegrityPolicy)_networkConnectionIntegrityPolicy
 {
-    return _websitePolicies->networkConnectionIntegrityEnabled();
+    _WKWebsiteNetworkConnectionIntegrityPolicy nCI = 0;
+    auto nCIOptions = _websitePolicies->networkConnectionIntegrityPolicy();
+
+    if (nCIOptions.contains(WebCore::NetworkConnectionIntegrity::Enabled))
+        nCI |= _WKWebsiteNetworkConnectionIntegrityPolicyEnabled;
+
+    if (nCIOptions.contains(WebCore::NetworkConnectionIntegrity::HTTPSFirst))
+        nCI |= _WKWebsiteNetworkConnectionIntegrityPolicyHTTPSFirst;
+
+    return nCI;
 }
 
-- (void)_setNetworkConnectionIntegrityEnabled:(BOOL)networkConnectionIntegrityEnabled
```

Comment:
I think you'll want to keep `-_setNetworkConnectionIntegrityEnabled:` around, for bincompat with clients that have already adopted this SPI. Perhaps this could just set the `enabled` flag?